.cbmc-batch/jobs/aws_cryptosdk_enc_ctx_clone/README.md

```diff
@@ -0,0 +1,18 @@
+# Memory safety proof for aws_cryptosdk_enc_ctx_clone
+
+This proof harness attains 95% code coverage.  The following comments explain
+why the uncovered lines of code are unreachable code.
+
+Some functions contain unreachable blocks of code:
+
+* `aws_hash_table_put`
+
+    * Edge cases never reached. was_created is always false and we never try and put an item in a full table
+
+* `aws_string_eq`
+
+    *  a !=b and neither a nor b are even NULL
```

Comment:
Same. Is this a limit of the harness, or does the code ensure this?

Reply:
Call to this function is preceded by a check that a is not NULL (b is not NULL by the set up of the loop inside which this function is called). This proof uses the representation of a hash_table given in aws_hash_table_no_slots_override.c (a hash-table that is not backed by any actual memory. It just takes non-det actions when given inputs) and otherwise the only assumption in the harness is for valid hash tables. I have updated the README with more information.